drivers/sensor/iis3dwb/iis3dwb.c

```diff
@@ -0,0 +1,404 @@
+/* ST Microelectronics IIS3DWB 3-axis accelerometer driver
+ *
+ * Copyright (c) 2020 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/iis3dwb.pdf
+ */
+
+#define DT_DRV_COMPAT st_iis3dwb
+
+#include <init.h>
+#include <sys/__assert.h>
+#include <sys/byteorder.h>
+#include <logging/log.h>
+#include <drivers/sensor.h>
+
+#include "iis3dwb.h"
+
+LOG_MODULE_REGISTER(IIS3DWB, CONFIG_SENSOR_LOG_LEVEL);
+
+/**
+ * iis3dwb_set_range_raw - set full scale range for acc
+ * @dev: Pointer to instance of struct device
+ * @range: Full scale range (2, 4, 8 and 16 G)
+ */
+static int iis3dwb_set_range_raw(const struct device *dev, uint8_t fs)
+{
+	int err;
+	struct iis3dwb_data *iis3dwb = dev->data;
+	const struct iis3dwb_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
```

Comment:
The cast isn't necessary

Reply:
correct

Reply:
It is DNM because during testing with SPI+DMA I realized that still I'm not able to stay in sync with 26K odr, and I need to spend some more time using the scope. But it might be just a matter of define a correct DT configuration; so I decided to open a PR anyway and feedback #40130.

Reply:
> 

Actually I have taken a look and I guess that we require this cast, otherwise following warning is generated because "const" is dropped:

```
/local/home/visconti/Projects/zephyrproject/zephyr/drivers/sensor/iis3dwb/iis3dwb.c:33:22: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
   33 |  stmdev_ctx_t *ctx = &cfg->ctx;
      |                      ^
```

Any suggestion on how to fix it easily? 
This issue was already opened in STdC drivers, but wrongly addressed , and I, who recently got the internal responsibility for them, may fix it in the next future. What I should do is to change all APIs in this way:

- int32_t lsm6dso_device_id_get(stmdev_ctx_t *ctx, uint8_t *buff);
+ int32_t lsm6dso_device_id_get(const stmdev_ctx_t *ctx, uint8_t *buff);

Reply:
I think you've answered your own question.

Reply:
> 

Yes, but it is a big change (more than 60 reporitories). And I need to plan it carefully later on.